src/HAEntity.hpp

```diff
@@ -6,16 +6,27 @@
 #include <iostream>
 #include <string>
 #include <sstream>
+#include "ext/magic_enum/include/magic_enum/magic_enum_all.hpp"
 
 using std::map;
 using std::string;
 
 using json = nlohmann::json;
 
+// TODO: we need to generate this to keep this in sync with HA.
+enum class EntityType
+{
+  light,
+  Switch,
+  climate,
+  OTHER
```

Comment:
```suggestion
  OTHER,
```

a trailing comma here is allowed, and makes diffs easier to read